examples/noise-js/index.js

```diff
@@ -0,0 +1,320 @@
+      });
+    },
+    async stop(contentTopic) {
+      if (intervals.has(contentTopic)) {
+        clearInterval(intervals.get(contentTopic));
+        intervals.delete(contentTopic);
+      }
+      if (subscriptions.has(contentTopic)) {
+        await subscriptions.get(contentTopic)();
+        subscriptions.delete(contentTopic);
+      } else {
+        console.log("Subscriptipon doesnt exist");
+      }
+    },
+  };
+
+  return [sender, responder];
+}
+
+async function confirmAuthCodeFlow(pairingObj) {
+  const authCode = await pairingObj.getAuthCode();
+  pairingObj.validateAuthCode(confirm("Confirm that authcode is: " + authCode));
+}
+
+async function hideQR() {
+  qrCanvas.remove();
+  qrUrl.remove();
+}
+
+async function disableUI() {
+  hideQR();
+  chatArea.remove();
+}
+
+// Function to update the fields to guide the user by disabling buttons.
+const updateFields = () => {
+  const readyToSend = nicknameInput.value !== "";
+  textInput.disabled = !readyToSend;
+  sendButton.disabled = !readyToSend;
+};
+
+// Protobuf
+const ProtoChatMessage = new protobuf.Type("ChatMessage")
+  .add(new protobuf.Field("timestamp", 1, "uint64"))
+  .add(new protobuf.Field("nick", 2, "string"))
+  .add(new protobuf.Field("text", 3, "bytes"));
+
+let messages = [];
+
+const updateMessages = () => {
+  messagesDiv.innerHTML = "<ul>";
+  messages.forEach((msg) => {
+    messagesDiv.innerHTML += `<li>${msg}</li>`;
+  });
+  messagesDiv.innerHTML += "</ul>";
+};
+
+const onMessage = (wakuMessage) => {
+  const { timestamp, nick, text } = ProtoChatMessage.decode(
+    wakuMessage.payload
+  );
+  const time = new Date();
+  time.setTime(Number(timestamp) * 1000);
+
+  messages.push(
+    `(${nick}) <strong>${utils.bytesToUtf8(
+      text
+    )}</strong> <i>[${time.toISOString()}]</i>`
+  );
+  updateMessages();
+};
+
+async function main() {
+  // Starting the node
+  const node = await createLightNode({
+    libp2p: {
+      peerDiscovery: [
+        new PeerDiscoveryStaticPeers(getPredefinedBootstrapNodes(Fleet.Test)),
+      ],
+    },
+  });
+
+  try {
+    await node.start();
+
+    await waitForRemotePeer(node, [Protocols.Filter, Protocols.LightPush]);
+
+    wakuStatusSpan.innerHTML = "connected";
+
+    const [sender, responder] = getSenderAndResponder(node);
+
+    const myStaticKey = noise.generateX25519KeyPair();
+
+    const pairingParameters = getPairingInfofromUrl();
+
+    const initiator = pairingParameters ? true : false;
+
+    let encoder;
+    let decoder;
+
+    if (initiator) {
+      console.log("Initiator");
+      qrCanvas.remove(); // Initiator does not require a QR code
+
+      const pairingObj = new noise.WakuPairing(
+        sender,
+        responder,
+        myStaticKey,
+        pairingParameters
+      );
+      const pExecute = pairingObj.execute(120000); // timeout after 2m
+
+      confirmAuthCodeFlow(pairingObj);
+
+      try {
+        handshakeStatusSpan.innerHTML = "executing handshake...";
```

Comment:
As soon as I open the page it says "executing handshake".

I'd expect it to wait for the handshake to be started by the other browser first. Or is it not possible?

In this case, it coul dbe changed to "Waiting for handshake".

Reply:
Agree, now it is aligned.